src/coreclr/jit/lsraloongarch64.cpp

```diff
@@ -0,0 +1,1343 @@
+            killMask = compiler->compHelperCallKillSet(CORINFO_HELP_STOP_FOR_GC);
+            BuildDefsWithKills(tree, 0, RBM_NONE, killMask);
+            break;
+
+        case GT_MUL:
+            if (tree->gtOverflow())
+            {
+                // Need a register different from target reg to check for overflow.
+                buildInternalIntRegisterDefForNode(tree);
+                setInternalRegsDelayFree = true;
+            }
+            FALLTHROUGH;
+
+        case GT_MOD:
+        case GT_UMOD:
+        case GT_DIV:
+        case GT_MULHI:
+        case GT_UDIV:
+        {
+            srcCount = BuildBinaryUses(tree->AsOp());
+            buildInternalRegisterUses();
+            assert(dstCount == 1);
+            BuildDef(tree);
+        }
+        break;
+
+        case GT_INTRINSIC:
+        {
+            noway_assert((tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Abs) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Ceiling) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Floor) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Round) ||
+                         (tree->AsIntrinsic()->gtIntrinsicName == NI_System_Math_Sqrt));
+
+            // Both operand and its result must be of the same floating point type.
+            GenTree* op1 = tree->gtGetOp1();
+            assert(varTypeIsFloating(op1));
+            assert(op1->TypeGet() == tree->TypeGet());
+
+            BuildUse(op1);
+            srcCount = 1;
+            assert(dstCount == 1);
+            BuildDef(tree);
+        }
+        break;
+
+#ifdef FEATURE_SIMD
+        case GT_SIMD:
+            srcCount = BuildSIMD(tree->AsSIMD());
+            break;
+#endif // FEATURE_SIMD
+
+#ifdef FEATURE_HW_INTRINSICS
+        case GT_HWINTRINSIC:
+            srcCount = BuildHWIntrinsic(tree->AsHWIntrinsic(), &dstCount);
+            break;
+#endif // FEATURE_HW_INTRINSICS
+
+        case GT_CAST:
+            assert(dstCount == 1);
+            srcCount = BuildCast(tree->AsCast());
+            break;
+
+        case GT_NEG:
+        case GT_NOT:
+            BuildUse(tree->gtGetOp1());
+            srcCount = 1;
+            assert(dstCount == 1);
+            BuildDef(tree);
+            break;
+
+        case GT_EQ:
+        case GT_NE:
+        case GT_LT:
+        case GT_LE:
+        case GT_GE:
+        case GT_GT:
+        case GT_JCMP:
+            srcCount = BuildCmp(tree);
+            break;
+
+        case GT_CKFINITE:
+            srcCount = 1;
+            assert(dstCount == 1);
+            buildInternalIntRegisterDefForNode(tree);
+            BuildUse(tree->gtGetOp1());
+            BuildDef(tree);
+            buildInternalRegisterUses();
+            break;
+
+        case GT_CMPXCHG:
+        {
+            NYI_LOONGARCH64("-----unimplemented on LOONGARCH64 yet----");
+            srcCount = 2;
```

Comment:
For `NYI_LOONGARCH64`, we should not set `sourceCount`.

Reply:
Yes, but here is just for avoiding the compiling error.